src/PasswordPolicy.php

```diff
@@ -0,0 +1,247 @@
+    }
+    return NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExpireDays(): int {
+    return floor($this->getExpireSeconds() / 86400);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExpireSeconds(): int {
+    return $this->expireSeconds;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExpireWarnSeconds(): int {
+    return $this->expireWarnSeconds;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExpireWarnDays(): int {
+    return floor($this->getExpireWarnSeconds() / 86400);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getExpiryWarningMessage(): ?string {
+    return $this->expiryWarningMessage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getMinimumRequiredConstraints(): int {
+    return $this->minimumRequiredConstraints;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPriority(): int {
+    return $this->role->getWeight();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRole(): RoleInterface {
+    return $this->role;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setExpireSeconds(int $seconds): PasswordPolicyInterface {
+    if ($seconds < 0) {
+      throw new PolicyInvalidArgumentException('The expiry given in seconds must be a positive integer or zero.');
+    }
+    $this->expireSeconds = $seconds;
+    // No need to store the expiry warning message if it's never to be shown.
+    if ($seconds == 0) {
+      $this->expiryWarningMessage = NULL;
+    }
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setExpireWarnSeconds(int $seconds): PasswordPolicyInterface {
+    if ($seconds < 0) {
+      throw new PolicyInvalidArgumentException('The expiry warning in seconds must be a positive integer or zero.');
+    }
+    $this->expireWarnSeconds = $seconds;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setExpiryWarningMessage(?string $message): PasswordPolicyInterface {
+    if ($this->expireSeconds > 0 && $this->expireWarnSeconds > 0 && !$message) {
+      throw new PolicyInvalidArgumentException('Expiry warning message cannot be empty for expiring passwords.');
+    }
+    $this->expiryWarningMessage = $message;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setMinimumRequiredConstraints(int $number): PasswordPolicyInterface {
+    if ($number < 0) {
+      throw new PolicyInvalidArgumentException('The minimum requirements cannot be less than zero.');
+    }
+    $this->minimumRequiredConstraints = $number;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConstraint($constraint): PasswordConstraintInterface {
+    return $this->constraints->get($constraint);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConstraints(): PasswordConstraintPluginCollection {
+    return $this->constraints;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function save(): void {
```

Comment:
`save()` is also a command method, something that is a value object (POJO) should not do. I would suggest moving it to the PP manager service.